lib/urlwatch/filters.py

```diff
@@ -855,3 +860,29 @@ def filter(self, data, subfilter):
             raise ImportError('Please install Pillow/PIL')
 
         return pytesseract.image_to_string(Image.open(io.BytesIO(data)), lang=language, timeout=timeout)
+
+
+class JQFilter(FilterBase):
+    """Parse, transform, and extract data from json as text using `jq`"""
+
+    __kind__ = 'jq'
+
+    __supported_subfilters__ = {
+        'query': 'jq query function to execute on data',
+    }
+
+    __default_subfilter__ = 'query'
+
+    def filter(self, data, subfilter):
+        try:
+            json.loads(data)
+        except ValueError:
+            raise ValueError('The url response contained invalid JSON')
+
+        if 'query' not in subfilter:
+            raise ValueError('{} filter needs a query'.format(self.__kind__))
+
+        if jq is None:
+            raise ImportError('Please install jq')
+
+        return jq.text(subfilter['query'], json.loads(data))
```

Comment:
One "optimization" here could be to store the result of `json.loads(data)` in line 878, and re-use the result in line 888 to not have to re-parse the JSON data.